packages/legend-query-builder/src/components/QueryBuilderParametersPanel.tsx

```diff
@@ -123,9 +114,52 @@ const VariableExpressionEditor = observer(
     const changeMultiplicity = (val: MultiplicityOption): void => {
       lambdaParameterState.changeMultiplicity(varState, val.value);
     };
+    const [parameterNameValue, setParameterNameValue] = useState(varState.name);
```

Comment:
hm, I think finding out in the alert is a bit late, why don't we do it the service pattern editor way? <----- potentially we can componentize that as well